src/components/calcite-rating/calcite-rating.tsx

```diff
@@ -0,0 +1,385 @@
+
+  /** the value of the rating component */
+  @Prop({ reflect: true }) value = 0;
+
+  /** is the rating component in a selectable mode */
+  @Prop({ reflect: true }) readOnly = false;
+
+  /** is the rating component in a selectable mode */
+  @Prop({ reflect: true }) disabled = false;
+
+  /** display rating value */
+  @Prop({ reflect: true }) displayValue = false;
+
+  /** optionally pass a number of previous ratings to display */
+  @Prop({ reflect: true }) count?: number;
+
+  /** optionally pass a cumulative average rating to display */
+  @Prop({ reflect: true }) average?: number;
+
+  //--------------------------------------------------------------------------
+  //
+  //  Events
+  //
+  //--------------------------------------------------------------------------
+
+  @Event() calciteRatingChange: EventEmitter;
+
+  //--------------------------------------------------------------------------
+  //
+  //  Event Listeners
+  //
+  //--------------------------------------------------------------------------
+
+  @Listen("calciteLabelFocus", { target: "window" }) handleLabelFocus(e) {
+    if (
+      hasLabel(e.detail.labelEl, this.el) &&
+      e.detail.interactedEl !== this.el &&
+      !this.el.contains(e.detail.interactedEl)
+    ) {
+      this.setFocus();
+    }
+  }
+
+  //--------------------------------------------------------------------------
+  //
+  //  Lifecycle
+  //
+  //--------------------------------------------------------------------------
+
+  componentDidLoad() {
+    this.ratingItems = this.el.shadowRoot.querySelectorAll(
+      "calcite-icon:not([data-partial='true'])"
+    );
+
+    this.determineInitialRating();
+  }
+
+  // --------------------------------------------------------------------------
+  //
+  //  Render Method
+  //
+  // --------------------------------------------------------------------------
+
+  render() {
+    this.dir = getElementDir(this.el);
+
+    const partialRatingStarContainer = (
+      <div
+        class="partial-rating-star-container"
+        data-partialhidden="false"
+        ref={(el) => (this.partialStarContainer = el)}
+      >
+        <calcite-icon
+          icon={this.selectedIconType}
+          scale={this.scale}
+          theme={this.theme}
+          data-partial="true"
+          ref={(el) => (this.partialStar = el)}
+        ></calcite-icon>
+      </div>
+    );
+
+    // ie11 safe version of array / spread / keys
+    const iconsToRender = [];
+    for (let i = 0; i < 5; i++) {
+      iconsToRender.push(
+        <calcite-icon
+          data-value={i + 1}
+          onClick={(e) => this.determineRatingPosition(e)}
+          onKeyDown={(e) => this.handleKeyDown(e)}
+          onMouseEnter={(e) => this.showSelectedIconOnHover(e)}
+          onTouchStart={(e) => this.showSelectedIconOnHover(e)}
+          onFocus={(e) => this.showSelectedIconOnHover(e)}
+          tabindex={!this.readOnly && !this.disabled ? 0 : null}
```

Comment:
Instead of querying for all the icons when the component is loaded, can you store the reference like so?

```tsx
ref={(el) => this.ratingItems[index] = el as HTMLCalciteIconElement}
```

Reply:
Maybe `ratingItems` could be a stateful prop incase the amount is configurable in the future?

Reply:
Yeah, good updates.